probe-rs/src/architecture/arm/sequences/stm32_armv6.rs

```diff
@@ -0,0 +1,211 @@
+//! Sequences for ARMv6 STM32s: STM32F0, STM32G0, STM32L0.
+//!
+//! This covers devices where DBGMCU is at 0x40015800 and has the DBG_STANDBY and DBG_STOP bits.
+
+use std::sync::Arc;
+
+use super::ArmDebugSequence;
+use crate::architecture::arm::{ap::MemoryAp, ApAddress, ArmError, ArmProbeInterface, DpAddress};
+
+/// Supported families for custom sequences on ARMv6 STM32 devices.
+pub enum Stm32Armv6Family {
+    /// STM32F0 family
+    F0,
+
+    /// STM32L0 family
+    L0,
+
+    /// STM32G0 family
+    G0,
+}
+
+/// Marker structure for ARMv6 STM32 devices.
+pub struct Stm32Armv6 {
+    family: Stm32Armv6Family,
+}
+
+impl Stm32Armv6 {
+    /// Create the sequencer for ARMv6 STM32 devices.
+    pub fn create(family: Stm32Armv6Family) -> Arc<Self> {
+        Arc::new(Self { family })
+    }
+}
+
+mod rcc {
+    use crate::architecture::arm::{memory::adi_v5_memory_interface::ArmProbe, ArmError};
+    use bitfield::bitfield;
+
+    /// The base address of the RCC peripheral
+    const RCC: u64 = 0x40021000;
+
+    bitfield! {
+        pub struct EnrF0(u32);
+        impl Debug;
+
+        pub u8, dbgen, enable_dbg: 22;
+    }
+
+    bitfield! {
+        pub struct EnrL0(u32);
+        impl Debug;
+
+        pub u8, dbgen, enable_dbg: 22;
+    }
+
+    bitfield! {
+        pub struct EnrG0(u32);
+        impl Debug;
+
+        pub u8, dbgen, enable_dbg: 27;
+    }
+
+    impl EnrF0 {
```

Comment:
Could we make those impls a macro? It bothers me a bit that it's so much copy paste :P But I see why we do it for typesafety :)

Reply:
Maybe we could make this a trait with default impl and a const associated constant `ADDRESS` then you just need an empty impl trait on each type :)